src/main/java/org/wildfly/security/auth/realm/LegacyPropertiesSecurityRealm.java

```diff
@@ -264,7 +266,22 @@ public void load(InputStream usersStream, InputStream groupsStream) throws IOExc
                             while (it.hasNext()) {
                                 int cp = it.next();
                                 if (cp == '\\' && it.hasNext()) { // escape
-                                    builder.appendCodePoint(it.next());
+                                    //might be regular escape of regex like characters \\t \\! or unicode \\uxxxx
+                                    int marker = it.next();
+                                    if(marker != 'u'){
+                                        builder.appendCodePoint(marker);
+                                    } else {
+                                        StringBuilder hex = new StringBuilder();
+                                        try{
+                                            hex.appendCodePoint(it.next());
+                                            hex.appendCodePoint(it.next());
+                                            hex.appendCodePoint(it.next());
+                                            hex.appendCodePoint(it.next());
+                                            builder.appendCodePoint((char)Integer.parseInt(hex.toString(),16));
+                                        } catch(NoSuchElementException nsee){
+                                            throw ElytronMessages.log.invalidUnicodeSequence(hex.toString());
```

Comment:
Checking `hasNext()` is preferred way, even through... maybe could be tolerated when an exception is result in the end...